src/reporting/clients/MappingsClient.ts

```diff
@@ -0,0 +1,641 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import type { AccessToken } from "@itwin/core-bentley";
+import { RequiredError } from "../interfaces/Errors";
+import type { EntityListIterator } from "../iterators/EntityListIterator";
+import { EntityListIteratorImpl } from "../iterators/EntityListIteratorImpl";
+import { Collection, getEntityCollectionPage } from "../iterators/IteratorUtil";
+import { OperationsBase } from "../OperationsBase";
+import { CalculatedProperty, CalculatedPropertyCollection, CalculatedPropertyCreate, CalculatedPropertySingle, CalculatedPropertyType, CalculatedPropertyUpdate } from "../interfaces/CalculatedProperties";
+import type { CustomCalculation, CustomCalculationCollection, CustomCalculationCreate, CustomCalculationSingle, CustomCalculationUpdate } from "../interfaces/CustomCalculations";
+import { DataType, GroupProperty, GroupPropertyCollection, GroupPropertyCreate, GroupPropertySingle, GroupPropertyUpdate } from "../interfaces/GroupProperties";
+import type { Group, GroupCollection, GroupCreate, GroupSingle, GroupUpdate } from "../interfaces/Groups";
+import type { Mapping, MappingCollection, MappingCopy, MappingCreate, MappingSingle, MappingUpdate } from "../interfaces/Mappings";
+import type { IMappingsClient } from "./IMappingsClient";
+
+export class MappingsClient extends OperationsBase implements IMappingsClient{
+  public async getMappings(accessToken: AccessToken, iModelId: string, top?: number): Promise<Mapping[]> {
+    const mappings: Array<Mapping> = [];
+    const mapIterator = this.getMappingsIterator(accessToken, iModelId, top);
+    for await(const map of mapIterator) {
+      mappings.push(map);
+    }
+    return mappings;
+  }
+
+  public getMappingsIterator(accessToken: AccessToken, iModelId: string, top?: number): EntityListIterator<Mapping> {
+    if(!this.topIsValid(top)) {
+      throw new RequiredError(
+        "top",
+        "Parameter top was outside of the valid range [1-1000] when calling getMappingsIterator."
+      );
+    }
+    let url = `${this.basePath}/datasources/imodels/${encodeURIComponent(iModelId)}/mappings`;
+    url += top ?  `/?%24top=${top}` : "";
```

Comment:
Replace `%24` with `$` and check if integration tests still pass after the change.

Reply:
changed to $, tests still pass